middleware/error_handling.go

```diff
@@ -12,10 +13,17 @@ func HandleErrors(next echo.HandlerFunc) echo.HandlerFunc {
 	return func(c echo.Context) error {
 		err := next(c)
 		if err != nil {
-			return c.JSON(
-				http.StatusInternalServerError,
-				util.ErrorDoc(fmt.Sprintf("Internal Server Error: %v", err.Error()), "500"),
-			)
+			var statusCode int
+			var message interface{}
+
+			if errors.Is(err, &util.NotFoundError{}) {
+				statusCode = http.StatusNotFound
+				message = util.ErrorDoc(err.Error(), "404")
+			} else {
+				statusCode = http.StatusInternalServerError
+				message = util.ErrorDoc(fmt.Sprintf("Internal Server Error: %v", err.Error()), "500")
```

Comment:
I like what you did here a lot, this is great. This way we can just let the error bubble up to the middleware which handles the return!